proj/src/main/webapp/script.js

```diff
@@ -205,3 +204,17 @@ function clickWithEnterKey() {
         }
     });
 }
+
+function createMessageWithTemplate(messageObj) {
+    const messageTemplate = document.getElementById('message-temp');
+    const message = messageTemplate.content.cloneNode(true);
+
+    const msgHeader = message.querySelector('.message-header');
+    const username = document.createElement('p');
```

Comment:
Why are you creating this in the script rather than including it in the template?